app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -152,7 +170,7 @@ private boolean isStudentImageValid(StudentImage studentImage){
     }
 
     // Delete StudentImageCollectionEvent and all StudentImages if a file doesn't exist anymore or the feature extraction failed for another reason
-    private void deleteStudentImagesRecursive(StudentImage studentImage, String reason){
+    private synchronized void deleteStudentImagesRecursive(StudentImage studentImage, String reason){
```

Comment:
Should we keep some data about this so that we can learn why a feature extraction failed for the collection of images?